src/test/java/com/google/api/codegen/CodegenTestUtil.java

```diff
@@ -72,4 +78,27 @@ public static ConfigProto readConfig(
 
     return (ConfigProto) configSource.getConfig();
   }
+
+  public static ServiceConfig readGRPCServiceConfig(
+      DiagCollector diagCollector, TestDataLocator testDataLocator, String serviceConfigFileName) {
+    URL serviceConfigUrl = testDataLocator.findTestData(serviceConfigFileName);
+
+    String serviceConfigPath = Objects.requireNonNull(serviceConfigUrl).getPath();
+
+    ServiceConfig.Builder b = ServiceConfig.newBuilder();
+
+    try {
+      FileReader file = new FileReader(serviceConfigPath);
+      JsonFormat.parser().merge(file, b);
```

Comment:
Is it possible to use `MuitiYamlReader`, similar to gapic config, as well? https://github.com/googleapis/gapic-generator/pull/2859/files#diff-d39112fadd134cc4a174510601615043L65

Reply:
Good idea. I would prefer to keep things in a JSON format however, because this is the format in which other systems that might use the gRPC ServiceConfig will expect it to be in. Thus, we aren't expecting anything different from other systems consuming the same config. (Disclaimer: these "other systems" might not exist yet, but the "how" they should function is well-defined by gRPC team).

I'm happy to look at making this more "formally" supported, but if possible, keeping it in JSON form makes life easier. 

WDYT?

Reply:
That makes sense. 

BTW will service configs be checked in into googleapis/googleapis? Also does artman need to be updated to pick up the service configs and pass that to gapic-generator?

Reply:
> Also does artman need to be updated to pick up the service configs and pass that to gapic-generator

most likely, yes! I need to explore this as i've never changed artman before. Any help/guidance is welcome!